src/runtime/security/security.thrift

```diff
@@ -0,0 +1,46 @@
+include "../../dsn.thrift"
+
+namespace cpp dsn.security
+
+// negotiation process:
+//
+//                       client                              server
+//                          | ---   SASL_LIST_MECHANISMS    --> |
+//                          | <-- SASL_LIST_MECHANISMS_RESP --- |
+//                          | --    SASL_SELECT_MECHANISMS  --> |
+//                          | <-- SASL_SELECT_MECHANISMS_OK --- |
+//                          |                                   |
+//                          | ---      SASL_INITIATE        --> |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---      SASL_RESPONSE        --> |
+//                          |                                   |
+//                          |              .....                |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---      SASL_RESPONSE        --> |
+//                          |                                   | (authentication will succeed
+//                          |                                   |  if all chanllenges passed)
+//                          | <--        SASL_SUCC          --- |
```

Comment:
`SASL_AUTH_FAIL` is not descript in this graph

Reply:
Yes, but it's hard to include both fail and success situations in one graph